### PR TITLE
Fix ../test/recipes/80-test_ssl_old.t with no-ec build

### PR DESCRIPTION
**Pull request by bernd-edlinger**

### CODE REVIEW

File: test/recipes/80-test_ssl_old.t [hunk 1]
```diff
@@ -456,7 +456,7 @@ sub testssl {
             # the default choice if TLSv1.3 enabled
             my $flag = $protocol eq "-tls1_3" ? "" : $protocol;
             foreach my $cipher (@{$ciphersuites{$protocol}}) {
-                if ($protocol eq "-ssl3" && $cipher =~ /ECDH/ ) {
+                if (($protocol eq "-ssl3" || $no_ec) && $cipher =~ /ECDH/ ) {
```
Comment (vdukhovni) [inline test/recipes/80-test_ssl_old.t]: This is not the right place to fix this.  Take a look above at line 414, and see what changes need to be made to the exclusions.

Comment (vdukhovni) [reply test/recipes/80-test_ssl_old.t]: The correct change is I think `s/kECDH/kECDHE/` on line 412.

Comment (bernd-edlinger) [reply test/recipes/80-test_ssl_old.t]: Oh, interesting...
Can you explain why
`openssl ciphers -s -tls1_2 ALL`
outputs ECDHE-ARIA256-GCM-SHA384 and ECDHE-ARIA128-GCM-SHA256
when configured with no-ec ???

Comment (bernd-edlinger) [reply test/recipes/80-test_ssl_old.t]: could it be that this is the root cause here ?
./config no-ec
=> openssl ciphers  ALL |sed 's/:/\n/g'|grep EC
ECDHE-ARIA256-GCM-SHA384
ECDHE-ARIA128-GCM-SHA256

**Comment by kroeckx** [comment]
On Tue, Dec 26, 2017 at 11:44:57AM -0800, Bernd Edlinger wrote:
> => openssl ciphers  ALL |sed 's/:/\n/g'|grep EC
> ECDHE-ARIA256-GCM-SHA384
> ECDHE-ARIA128-GCM-SHA256

Those ciphers should be inside an #ifndef OPENSSL_NO_EC in s3_lib.c

**Comment by kroeckx** [comment]
I wonder if we should also put the SSL_kECDHE define inside an
#ifndef OPENSSL_NO_EC, to prevent something like this.

**Comment by bernd-edlinger** [comment]
Boah, yes!
But funny other ECDHE ciphers there are not #ifndef OPENSSL_NO_EC
And I wonder if the code at line 412 is not just papering over another problem...
Now this issue _has_ my attention.

File: test/recipes/80-test_ssl_old.t (hunk 1 above)
Comment (richsalz) [reply test/recipes/80-test_ssl_old.t]: Yes, @bernd-edlinger I believe you're right.  Paging @paulidale

Comment (bernd-edlinger) [reply test/recipes/80-test_ssl_old.t]: how is the string "-kECDHe" in line 399 supposed to work?
from grep it looks like not implemented:

```
grep -r kECDHe *
CHANGES:         kECDHe   - ECDH cert, signed with ECDSA
include/openssl/ssl.h:# define SSL_TXT_kECDHe          "kECDHe"
test/recipes/80-test_ssl_old.t:        my $ciphers = "-EXP:-PSK:-SRP:-kDH:-kECDHe";
```

**Comment by bernd-edlinger** [comment]
See #4981 for an alternate approach on fixing this.

**Pull request closed without merging**